### PR TITLE
JCLOUDS-1274: Delete managed disks when cleaning up VM resources

### PR DESCRIPTION
**Pull request by nacx**
https://issues.apache.org/jira/browse/JCLOUDS-1274

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CleanupResources.java [hunk 1]
```diff
@@ -95,16 +103,40 @@ public void cleanupVirtualMachineNICs(VirtualMachine virtualMachine) {
...
+      return all(filter(deleteJobs, notNull()), resourceDeleted);
+   }
+   
+   @Nullable
+   private URI deleteManagedDisk(@Nullable ManagedDiskParameters managedDisk) {
+      if (managedDisk != null) {
+         IdReference diskRef = IdReference.create(managedDisk.id());
+         logger.debug(">> deleting managed disk %s...", diskRef.name());
+         return api.getDiskApi(diskRef.resourceGroup()).delete(diskRef.name());
+      }
```
Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CleanupResources.java]: maybe we can `log.warn` that the disk didn't get deleted?

**Comment by andreaturli** [comment]
minor comment, not tested, but looks good to me

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CleanupResources.java (hunk 1 above)
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CleanupResources.java]: Done!

**Comment by andreaturli** [comment]
cool +1

**Comment by nacx** [comment]
Squashed and pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/004fbcf8) and [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/9dc23402). Thanks for the reviews @andreaturli!

**Pull request closed without merging**